File: source-connector/src/main/java/sourceconnector/domain/connect/Task.java
package sourceconnector.domain.connect;

import offsetmanager.domain.file.FileKey;

import java.util.Map;
import java.util.concurrent.Callable;

public interface Task<T> extends Callable<T> {
  int getIndex();

  /**
   * Assign the file paths handled by this task
   */
  void assign(Map<FileKey, Long> offsetMap);
}

File: source-connector/src/main/kotlin/sourceconnector/domain/connect/Task.kt
package sourceconnector.domain.connect

import offsetmanager.domain.file.FileKey
import java.util.concurrent.Callable

interface Task<T> : Callable<T> {
  val index: Int

  /**
   * Assign the file paths handled by this task
   */
  fun assign(offsetMap: Map<FileKey, Long>)
}
